File: frontend/src/types/operationalPublication.ts
export type OperationalPublicationStatus = "DRAFT" | "READY" | "PUBLISHED" | "ARCHIVED";
export type ChecklistItemLevel = "OK" | "WARN" | "ERROR" | "PENDING";

export interface ChecklistItem {
  key: string;
  title: string;
  level: ChecklistItemLevel;
  detail: string;
  blocking: boolean;
}

export interface OperationalPublicationChecklist {
  items: ChecklistItem[];
  ready: boolean;
  has_errors: boolean;
  has_warnings: boolean;
  can_publish_with_risk: boolean;
}

export interface OperationalPublicationPublic {
  id: number;
  service_scale_id: number;
  scale_date: string;
  publication_number: number;
  version: number;
  status: OperationalPublicationStatus;
  created_by_id: number;
  created_by_label: string | null;
  published_by_id: number | null;
  published_by_label: string | null;
  published_at: string | null;
  generated_message: string | null;
  generated_pdf: string | null;
  change_summary: string | null;
  publish_reason: string | null;
  risk_acknowledged: boolean;
  previous_publication_id: number | null;
  checklist: OperationalPublicationChecklist | null;
  created_at: string;
  updated_at: string;
}

export interface OperationalPublicationAuditPublic {
  id: number;
  action: string;
  actor_id: number;
  actor_label: string | null;
  details: string | null;
  created_at: string;
}

export interface OperationalPublicationDetail extends OperationalPublicationPublic {
  snapshot: Record<string, unknown> | null;
  audits: OperationalPublicationAuditPublic[];
}

export interface OperationalPublicationHistoryItem {
  id: number;
  service_scale_id: number;
  scale_date: string;
  publication_number: number;
  version: number;
  status: OperationalPublicationStatus;
  published_by_label: string | null;
  published_at: string | null;
  publish_reason: string | null;
  change_summary: string | null;
  risk_acknowledged: boolean;
}

export interface OperationalPublicationHistoryResponse {
  items: OperationalPublicationHistoryItem[];
  total: number;
}

export interface OperationalPublicationCenterDay {
  scale_date: string;
  service_scale_id: number | null;
  scale_title: string | null;
  scale_status: string | null;
  active_publication: OperationalPublicationPublic | null;
  checklist: OperationalPublicationChecklist | null;
  latest_published: OperationalPublicationHistoryItem | null;
}

File: backend/schemas/operational_publication.py
"""Schemas do domínio OperationalPublication."""

from __future__ import annotations

from datetime import date, datetime
from enum import Enum

from pydantic import BaseModel, Field

from models.operational_publication import (
    OperationalPublicationAuditAction,
    OperationalPublicationStatus,
)


class ChecklistItemLevel(str, Enum):
    OK = "OK"
    WARN = "WARN"
    ERROR = "ERROR"
    PENDING = "PENDING"


class ChecklistItem(BaseModel):
    key: str
    title: str
    level: ChecklistItemLevel
    detail: str
    blocking: bool = False


class OperationalPublicationChecklist(BaseModel):
    items: list[ChecklistItem]
    ready: bool
    has_errors: bool
    has_warnings: bool
    can_publish_with_risk: bool


class OperationalPublicationCreateDraft(BaseModel):
    service_scale_id: int
    scale_date: date | None = None


class OperationalPublicationPublishRequest(BaseModel):
    acknowledge_risks: bool = False
    reason: str | None = Field(default=None, max_length=512)


class OperationalPublicationAuditPublic(BaseModel):
    id: int
    action: OperationalPublicationAuditAction
    actor_id: int
    actor_label: str | None = None
    details: str | None = None
    created_at: datetime


class OperationalPublicationPublic(BaseModel):
    id: int
    service_scale_id: int
    scale_date: date
    publication_number: int
    version: int
    status: OperationalPublicationStatus
    created_by_id: int
    created_by_label: str | None = None
    published_by_id: int | None = None
    published_by_label: str | None = None
    published_at: datetime | None = None
    generated_message: str | None = None
    generated_pdf: str | None = None
    change_summary: str | None = None
    publish_reason: str | None = None
    risk_acknowledged: bool = False
    previous_publication_id: int | None = None
    checklist: OperationalPublicationChecklist | None = None
    created_at: datetime
    updated_at: datetime


class OperationalPublicationDetail(OperationalPublicationPublic):
    snapshot: dict | None = None
    audits: list[OperationalPublicationAuditPublic] = Field(default_factory=list)


class OperationalPublicationHistoryItem(BaseModel):
    id: int
    service_scale_id: int
    scale_date: date
    publication_number: int
    version: int
    status: OperationalPublicationStatus
    published_by_label: str | None = None
    published_at: datetime | None = None
    publish_reason: str | None = None
    change_summary: str | None = None
    risk_acknowledged: bool = False


class OperationalPublicationHistoryResponse(BaseModel):
    items: list[OperationalPublicationHistoryItem]
    total: int


class OperationalPublicationCenterDay(BaseModel):
    scale_date: date
    service_scale_id: int | None
    scale_title: str | None
    scale_status: str | None
    active_publication: OperationalPublicationPublic | None
    checklist: OperationalPublicationChecklist | None
    latest_published: OperationalPublicationHistoryItem | None
